### ISSUE TITLE
Gráficos gerados pelo pacote estão diferentes daqueles mostrados no cookbook

### ISSUE THREAD

**Issue by palmaresk8** [ipeadata-lab/ipeaplot#41]
Olá. Estou testando o pacote aqui. A primeira coisa que percebi é que rodando o código do cookbook eu obtenho alguns gráficos com labels do eixo x deslocados em relação às barras que os delimitam (os 'rugs'). Além disso, os números não aparecem como números inteiros.
![Grafico 1](https://github.com/ipeadata-lab/ipeaplot/assets/6098533/ad77c9ca-f303-4739-bf94-12328e2f0a61)
![Grafico 2](https://github.com/ipeadata-lab/ipeaplot/assets/6098533/7232057e-a83d-4374-aa2f-0042fa9b586a)
![Grafico 5](https://github.com/ipeadata-lab/ipeaplot/assets/6098533/cdd40dd5-f044-4f45-a39c-f6dce0cf6187)

Outra questão não relacionada à acima é se títulos, subtítulos e notas devem fazer parte do gráfico. Em publicações recentes no portal do Ipea (vide [](https://repositorio.ipea.gov.br/bitstream/11058/12698/4/TD_2951_web.pdf)), me parece que não, que esses elementos são somente textuais e não fazem parte da "figura" do gráfico.
![2024-03-07_115449](https://github.com/ipeadata-lab/ipeaplot/assets/6098533/45b4762d-5cd3-4cce-bff2-c5dd8c70b7c7)

**Comment by palmaresk8** [comment]
@cavalcanti1985 , sim, eu entendo que o manual diga isso, e o estilo gráfico das publicações até então indicava que título, notas etc deviam fazer parte da "figura" do gráfico. Mas, se você perceber em publicações recentes, houve uma mudança estética que creio não está "escrita" ainda, no qual visualmente parece que esses elementos não fazem mais parte da "figura" do gráfico. Essa mudança de design é bem recente, tanto que uma ou outra publicação que estava há mais tempo no editorial ainda está saindo com o estilo antigo. Mas o link que eu coloquei é um bom exemplo [(esse aqui)](https://repositorio.ipea.gov.br/bitstream/11058/12698/4/TD_2951_web.pdf). Enfim... só pontuando se vale a pena conferir com o editorial sobre isso. De todo modo, isso não exige uma alteração do pacote, pois se a pessoa quiser ela pode fazer o gráfico sem os títulos e notas.

**Comment by PedroJorge7** [comment]
Fizemos alguns testes para corrigir os problemas apontados pelo Fabio e acredito que eles foram resolvidos. Se puderem depois também testar para confirmar que não há algum caso em que a nova solução não funcione.

![grafico2](https://github.com/ipeadata-lab/ipeaplot/assets/52927575/500d3e1d-2185-4859-b835-ca2645b752bb)
![grafico5](https://github.com/ipeadata-lab/ipeaplot/assets/52927575/5926f5c3-5f40-4433-a7fc-33fdd34b0340)
![grafico1](https://github.com/ipeadata-lab/ipeaplot/assets/52927575/2cc7bf3a-8a23-406a-895c-deac8364d11a)

**Comment by palmaresk8** [comment]
Olá @PedroJorge7 . Os gráficos 1 e 2 estão iguais. O gráfico 5 está corrigido, mas usando o código do cookbook os labels do eixo x estão aparecendo a cada dois anos. (lembrando aqui que eu não testei todos os gráficos, eu só coloquei o 1, 2, e 5 como exemplos). Usando o commit 39b6aeb.
Um ponto a se notar é o aparecimento de alguns warnings. No gráfico 1 apareceu o seguinte:

> Warning messages:
1: In rbind(test, temp) :
  number of columns of result is not a multiple of vector length (arg 2)
2: In ggplot_add.scale_auto_ipea(object, p, objectname) :
  Due to the existence of a bar chart, the `expand_y_limit` argument will be converted to FALSE
3: In ggplot_add.scale_auto_ipea(object, p, objectname) :
  Due to the existence of a bar chart, the `x_breaks ` argument will be converted to the number of options available

No gráfico 5 apareceu:

> Warning messages:
1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was generated. 
2: In rbind(test, temp) :
  number of columns of result is not a multiple of vector length (arg 2)

**Comment by palmaresk8** [comment]
Ok, testando aqui todos os gráficos do cookbook. Além desses warnings que aparecem, e do gráfico 5 estar com os labels x sendo mostrados a cada dois anos, o Gráfico 8 foi gerado de forma errada.

![Grafico_5](https://github.com/ipeadata-lab/ipeaplot/assets/6098533/f8e69e7c-fd0a-421b-bd61-174e8ca2d6b5)
Nota: esse gráfico 5 está ok, somente os labels x não estão "exatamente" como mostra no cookbook.

![Grafico_8](https://github.com/ipeadata-lab/ipeaplot/assets/6098533/52954c92-358e-4eb7-bdba-128ce72c9ecf)

O gráfico 8 está errado. Ao executar o código do cookbook, apareceu as seguintes mensagens:

> Scale for x is already present.
Adding another scale for x, which will replace the existing scale.
Warning messages:
1: In rbind(test, temp) :
  number of columns of result is not a multiple of vector length (arg 2)
2: In ggplot_add.scale_auto_ipea(object, p, objectname) :
  Due to the existence of a bar chart, the `expand_y_limit` argument will be converted to FALSE
3: In ggplot_add.scale_auto_ipea(object, p, objectname) :
  Due to the existence of a bar chart, the `x_breaks ` argument will be converted to the number of options available

**Comment by lucasmation** [comment]
mais uma coisa, @PedroJorge7 , ao fazer o commit voce deve fazer referencia ao numero do issue. Para evitar este tipo de duvida que o Fabio está tendo)

**Comment by rafapereirabr** [comment]
Muito bom ver toda essa atividade aqui, mas algumas questões levantadas aqui mereciam ser tratadas em issues separados para facilitar  o tratamento de cada peoblema
